### PR TITLE
Refactor miopen conv_transpose operator

### PR DESCRIPTION
**Pull request by rohithkrn**
Refactor miopen conv_transpose operator to use miopenConvolution mode.

### CODE REVIEW

**Comment by rohithkrn** [comment]
@bddppq @petrex

File: caffe2/operators/hip/conv_transpose_op_miopen.hip [hunk 1]
```diff
@@ -105,28 +105,28 @@ class MIOPENConvTransposeGradientOp final : public MIOPENConvTransposeOpBase {
             OperatorBase::GetSingleArgument<bool>("bestAlgoFound", false)),
         bestWeightAlgoFound_(
             OperatorBase::GetSingleArgument<bool>("bestAlgoFound", false)),
+        fwdAlgo_(miopenConvolutionFwdAlgoGEMM),
```
Comment (bddppq) [inline caffe2/operators/hip/conv_transpose_op_miopen.hip]: Here `fwdAlgo_` is initialized before `bwdWeiAlgo_` while in line 140 and 141 they are declared in the reversed order

Comment (rohithkrn) [reply caffe2/operators/hip/conv_transpose_op_miopen.hip]: Fixed in lines 140 and 141.

**Review by bddppq — APPROVED** [review]

**Review by bddppq — CHANGES_REQUESTED** [review]
Put it on hold until new miopen release @rohithkrn @petrex

**Comment by bddppq** [comment]
@petrex @rohithkrn Is this one ready now as we have updated to rocm 2.1?

**Comment by petrex** [comment]
We have made miopen backward compatible so the PR is not required. Thx

Get Outlook for iOS<https://aka.ms/o0ukef>

________________________________

**Pull request closed without merging**